moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp

```diff
@@ -271,11 +259,9 @@ bool pilz_industrial_motion_planner::generateJointTrajectory(
 
       if (time_iter != time_samples.begin() && time_iter != time_samples.end() - 1)
       {
-        double joint_velocity =
-            (ik_solution.at(joint_name) - ik_solution_last.at(joint_name)) / duration_current_sample;
+        double joint_velocity = (ik_solution.at(joint_name) - ik_solution_last.at(joint_name)) / sampling_time;
         point.velocities.push_back(joint_velocity);
-        point.accelerations.push_back((joint_velocity - joint_velocity_last.at(joint_name)) /
-                                      (duration_current_sample + sampling_time) * 2);
```

Comment:
Can you please explain, why you simplified the code here?
Why was division by `(duration_current_sample + sampling_time) / 2)` required before, but now only by `sampling_time`?

Reply:
I think previously it was the time between the mid-points of the two segments.
 - If neither segment was the end, then `duration_current_sample` is just `sampling_time`, so this just became `(sampling_time + sampling_time) / 2 = sampling_time`.
 - For the final segment, `duration_current_sample` was small, so this would give a smaller interval.

Now that the segments are all evenly spaced, this can always just be `actual_sampling_time`.